### ISSUE TITLE
environments: not all environment variables point to view directory 

### ISSUE THREAD

**Issue by glennpj** [LLNL/spack#35661]
### Steps to reproduce

When creating a spack environment, the contents of environment variables should point to a directory in the environment view. This does not happen for all environment variables.
```console
$ spack env create relion
==> Created environment 'relion' in /opt/packages/gpjohnsn/spack/var/spack/environments/relion
==> You can activate this environment with:
==>   spack env activate relion
```
```console
$ spack -e relion add relion
==> Adding relion to environment relion
$ spack -e relion install
...
==> Updating view at /opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view
```
It is expected that all environment variables will point to directories in the view.
```
$ spack env activate --sh relion
export SPACK_ENV=/opt/packages/gpjohnsn/spack/var/spack/environments/relion;
alias despacktivate='spack env deactivate';
export ACLOCAL_PATH=/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view/share/aclocal;
export CMAKE_PREFIX_PATH=/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view;
export GI_TYPELIB_PATH=/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/gtkplus-3.24.29-fbyziq3mmv4lag2ttq4i67bcfelhnc62/lib/girepository-1.0:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/pango-1.50.7-edfi3sd4inuup5egiyrnx6vmwxwzzooe/lib/girepository-1.0:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/harfbuzz-5.3.1-td6tdryv66h5oablzvnoqa7q4w3jiwfc/lib/girepository-1.0:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/gdk-pixbuf-2.42.9-bngu4ukea3ts4x5ku6r363w7p6islk25/lib/girepository-1.0:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/atk-2.36.0-iyezaoztww73sujauazafutwry4kvlpe/lib/girepository-1.0:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/at-spi2-core-2.40.1-cmsxqqo2pnnv2tin4waw22rwc5rzrcon/lib/girepository-1.0:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/gobject-introspection-1.72.0-edrodugefhy55yqarfp3zwmwpskeuisi/lib/girepository-1.0;
export MANPATH=/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view/share/man:/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view/man:/usr/share/lmod/lmod/share/man:/opt/puppetlabs/puppet/share/man:.:;
export PATH=/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view/bin:/opt/packages/gpjohnsn/spack/bin:/usr/share/lmod/lmod/libexec:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/opt/puppetlabs/bin:/Users/gpjohnsn/.local/bin:/Users/gpjohnsn/bin;
export PKG_CONFIG_PATH=/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view/share/pkgconfig:/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view/lib64/pkgconfig:/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view/lib/pkgconfig;
export XDG_DATA_DIRS=/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/gtkplus-3.24.29-fbyziq3mmv4lag2ttq4i67bcfelhnc62/share:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/librsvg-2.51.0-tbwo2emjyuolqmmb4asdgl7rmr7wfd36/share:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/pango-1.50.7-edfi3sd4inuup5egiyrnx6vmwxwzzooe/share:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/harfbuzz-5.3.1-td6tdryv66h5oablzvnoqa7q4w3jiwfc/share:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/gdk-pixbuf-2.42.9-bngu4ukea3ts4x5ku6r363w7p6islk25/share:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/shared-mime-info-1.9-omxxccezjbrnakby3u5ytgfaplzghv7y/share:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/atk-2.36.0-iyezaoztww73sujauazafutwry4kvlpe/share:/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/gobject-introspection-1.72.0-edrodugefhy55yqarfp3zwmwpskeuisi/share;
export XLOCALEDIR=/opt/packages/gpjohnsn/spack/opt/spack/linux-centos7-cascadelake/gcc-9.5.0/libx11-1.7.0-dcwicileppi5cmoex4cjblmjttxhgkvm/share/X11/locale;
```
The variables for the following point to directories in the view.

- `ACLOCAL_PATH`
- `CMAKE_PREFIX_PATH`
- `MANPATH`
- `PATH`
- `PKG_CONFIG_PATH`

However, the following point to package directories.

- `GI_TYPELIB_PATH`
This should point to `/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view/lib/girepository-1.0`, which does exist in the view.
- `XDG_DATA_DIRS`
This should point to `/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view/share`, which does exist in the view.
- `XLOCALEDIR`
This should point to `/opt/packages/gpjohnsn/spack/var/spack/environments/relion/.spack-env/view/share/X11/locale`, which does exist in the view.

### Error message

There is no additional output when running 
```
$ spack  --debug --stacktrace env activate --sh relion
```

### Information on your system

```console
$ spack debug report
* **Spack:** 0.20.0.dev0
* **Python:** 3.6.8
* **Platform:** linux-centos7-cascadelake
* **Concretizer:** clingo
```

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
I think part of this is just package recipes, like `XLOCALE_DIR`, other parts may be core Spack. Triaging as impact-low as it's worth getting rid of the second case.

**Comment by haampie** [comment]
Fixed by #41737
